### ISSUE TITLE
ClassLoaderHandler for Quarkus RuntimeClassLoader

### ISSUE THREAD

**Issue by mcollovati** [classgraph/classgraph#405]
I recently did some experimenting with [quarkus](https://quarkus.io/) on a project that also uses classgraph. To make it work I wrote a custom `ClassLoaderHandler` for quarkus `RuntimeClassLoader`.

I don't know if the implementation is complete and if it's of interest to someone else,
so I'll post it here; I'd be happy to possibly open a PR.

```java
public class QuarkusClassLoaderHandler implements ClassLoaderHandler {

    private QuarkusClassLoaderHandler() {
    }

    public static boolean canHandle(final Class<?> classLoaderClass, final LogNode log) {
        return "io.quarkus.runner.RuntimeClassLoader".equals(classLoaderClass.getName());
    }

    public static void findClassLoaderOrder(final ClassLoader classLoader, final ClassLoaderOrder classLoaderOrder,
                                            final LogNode log) {
        classLoaderOrder.delegateTo(classLoader.getParent(), /* isParent = */ true, log);
        classLoaderOrder.add(classLoader, log);
    }

    public static void findClasspathOrder(final ClassLoader classLoader, final ClasspathOrder classpathOrder,
                                          final ScanSpec scanSpec, final LogNode log) {

        Map<String, Path> applicationClasses = (Map<String, Path>) ReflectionUtils.getFieldVal(classLoader, "applicationClasses", false);
        if (applicationClasses != null) {
            for (Path path : applicationClasses.values()) {
                classpathOrder.addClasspathEntry(path.toUri(), classLoader, scanSpec, log);
            }
        }
        Collection<Path> applicationClassDirectories = (Collection<Path>)ReflectionUtils.getFieldVal(classLoader, "applicationClassDirectories", false);
        if (applicationClassDirectories != null) {
            for (Path path : applicationClassDirectories) {
                classpathOrder.addClasspathEntryObject(path.toUri(), classLoader, scanSpec, log);
            }
        }
    }

}
```

**Comment by lukehutch** [comment]
Awesome, thanks for this. If it works for you, that's good enough to push out as a first iteration. If there are further changes needed to support different Quarkus versions or something, we can continue to improve this later. Please do submit a PR.

**Comment by lukehutch** [comment]
I figured I'd save you the trouble of creating a PR. Commited, and released in version 4.8.66. Thanks for the contribution!

**Comment by lukehutch** [comment]
Sonatype was having an issue publishing releases, so I had to delete the tag and re-tag 4.8.66. That means your second improved version of the `ClassLoaderHandler` will be in 4.8.66. Unfortunately I can't publish until Sonatype fixes their issue however.

**Comment by lukehutch** [comment]
Released in 4.8.66. Thanks again!
